src/components/todo/todo.tsx

```diff
@@ -0,0 +1,211 @@
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { ChangeEvent, useEffect, useRef, useState } from 'react';
+
+type Props = {
+  list: Todo[];
+  todo: Todo;
+  idTodo: number;
+  onDelete: (id: number) => Promise<void>;
+  onUpdate: (todo: Todo) => Promise<void>;
+  onLoading: (status: boolean) => void;
+  onError: (status: string) => void;
+  onIdTodo: (id: number) => void;
+};
+
+export const ToDo: React.FC<Props> = ({
+  list,
+  todo,
+  idTodo,
+  onDelete,
+  onUpdate,
+  onLoading,
+  onError,
+  onIdTodo,
+}) => {
+  const { title, userId, id, completed } = todo;
+  const [updateTodo, setUpdateTodo] = useState<Todo | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const titleField = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    titleField.current?.focus();
+  });
+
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    onError('');
+    onIdTodo(id);
+
+    if (updateTodo === null) {
+      setIsSubmitting(false);
+
+      return;
+    }
+
+    if (!updateTodo.title.trim()) {
+      onIdTodo(updateTodo.id);
+      onDelete(updateTodo.id)
+        .then(() => {
+          setUpdateTodo(null);
+          onIdTodo(updateTodo.id);
+        })
+        .finally(() => {
+          onLoading(false);
+          onIdTodo(0);
+        });
+    } else {
+      onIdTodo(0);
+      const todoActual = list.find(item => item.id === updateTodo.id);
+
+      if (todoActual && updateTodo.title === todoActual.title) {
+        setUpdateTodo(null);
+        setIsSubmitting(false);
+
+        return;
+      }
+
+      onIdTodo(updateTodo.id);
+      onLoading(true);
+      onUpdate({ ...updateTodo, title: updateTodo.title.trim() })
+        .then(() => {
+          setUpdateTodo(null);
+          onIdTodo(updateTodo.id);
+        })
+        .finally(() => {
+          onLoading(false);
+          onIdTodo(0);
+        });
+    }
+
+    setTimeout(() => {
+      setIsSubmitting(false);
+    }, 1000);
+  };
+
+  const handleChecked = (post: Todo) => {
+    onLoading(true);
+    onIdTodo(post.id);
+
+    const updateCompletedTodo = {
+      ...post,
+      completed: !post.completed,
+    };
+
+    onUpdate(updateCompletedTodo)
+      .then(() => setUpdateTodo(null))
+      .finally(() => {
+        onLoading(false);
+        onIdTodo(0);
+      });
+  };
```

Comment:
You can not wait for post as an argument and work with your current todo
```suggestion
  const handleChecked = () => {
    onLoading(true);
    onIdTodo(id);

    const updateCompletedTodo = {
      ...todo,
      completed: !completed,
    };

    onUpdate(updateCompletedTodo)
      .then(() => setUpdateTodo(null))
      .finally(() => {
        onLoading(false);
        onIdTodo(0);
      });
  };
```